Restructure Synchronizing Template Repositories (#2414)
theforeman/foreman-documentation: guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc

----

.Procedure
. Create the directory where templates are stored and apply appropriate permissions and SELinux context:
. In `/var/lib/foreman`, create a directory for storing templates:
+
[subs="+quotes"]
----
# mkdir -p _/usr/share/templates_dir/_

> # chown foreman _/usr/share/templates_dir/_

Reviewer: This is still needed: if you create it as  root then Foreman can't write to it.

Author: Thanks, restored.

> # mkdir /var/lib/foreman/_My_Templates_Dir_

Reviewer: Is this even needed? I'd think the plugin creates this by itself (if it has permissions).

cc @theforeman/foreman_templates

Reviewer 2: It depends on what values you provide in the export form.

`Repo` can be either full path or a "root" path. In this case it can be both `/var/lib/foreman/_My_Templates_Dir_` and `/var/lib/foreman`. In this case if `_My_Templates_Dir` doesn't exist, it won't be created.
`Dirname` can be omitted or specified. If specified then the plugin will create this directory in `Repo` root.

TL;DR:
`_My_Templates_Dir_` won't be created unless it's specified as `Dirname` in export form.

> [NOTE]
> ====
> You can place your templates to a custom directory outside `/var/lib/foreman`, but you have to set the `foreman_lib_t` SELinux label to that directory.
> ====

Reviewer 3: ```suggestion
If you want to place your templates to a custom directory outside `/var/lib/foreman`, you have to set the `foreman_lib_t` SELinux label to that directory.
```

I am unsure if a "note" admonition is required here.

> 

Reviewer 3: ```suggestion
```

. Change the *Repo* setting on the *TemplateSync* tab to match the export directory `/var/lib/foreman/_My_Templates_Dir_/`.
. Export templates from your {ProjectServer} to a local directory:
+
[options="nowrap", subs="verbatim,quotes,attributes"]
----
# curl -H "Accept:application/json" \
-H "Content-Type:application/json" \
-u _login_:__password__ \

> -k https://_{foreman-example-com}/api/v2/templates/export \

Reviewer: Just below this is `-X POST \` but I think that should be `-X POST`.

Author: Thanks, fixed.
